### DISCUSSION TITLE
Choosing the right record for the Event

### DISCUSSION THREAD

**Discussion by adrienpoly** [excid3/noticed#471] [Q&A] [answered]
We have migrated from V1 to V2 and overall loving it. Great job @excid3 and team

There is one part where I am a bit confused is for choosing the  right record for the Noticed::Event

Here is a simplified version of one of our notification 

We have Teachers having Students. Students write Stories and Teachers correct their Stories

We have a `StudentStory` model that is being corrected by a Teacher. Once corrected we create a notification

```ruby
StudentStory::CorrectedNotifier.with(record: ...).deliver(student)
```

For the record I wonder if I should give the `studend_story` object or the `student` 

My first idea was to use the student_story as the record but then I thought maybe using the student would simplify some queries down the road? 

Do you have any suggesting on how to best choose the record?

Thanks

**Comment by excid3** [comment, marked as answer]
Yep, sounds right to me. Record should be whatever object the notification is primarily about because you'll need it for rendering.
